### PR TITLE
`spectral_factor_firstsolar`: rename `pw` to `precipitable_water`

### PR DESCRIPTION
**Pull request by kandersolar**
- ~[ ] Closes #xxxx~
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - ~[ ] Tests added~
 - ~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - ~[ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

`spectral_factor_firstsolar` (formerly `first_solar_spectral_correction`) has a `pw` parameter.  This PR renames it to `precipitable_water` for consistency with the rest of pvlib.

### CODE REVIEW

File: pvlib/spectrum/mismatch.py [hunk 1]
```diff
@@ -317,6 +310,14 @@ def spectral_factor_firstsolar(pw, airmass_absolute, module_type=None,
...
+        minimum atmospheric precipitable water. Any pw value lower than min_pw
+        is set to min_pw to avoid model divergence. [cm]
+
+    max_pw : float, default 8
+        maximum atmospheric precipitable water. Any pw value higher than max_pw
+        is set to NaN to avoid model divergence. [cm]
+
     Returns
     -------
     modifier: array-like
```
Comment (AdamRJensen) [inline pvlib/spectrum/mismatch.py:323]: There's a typo below in the sentence "which is can be multiplied" (I can't edit that part though)

File: pvlib/spectrum/mismatch.py [hunk 2]
```diff
@@ -317,6 +310,14 @@ def spectral_factor_firstsolar(pw, airmass_absolute, module_type=None,
         modules with very similar quantum efficiency should be similar,
         in most cases limiting the need for module specific coefficients.
```
Comment (AdamRJensen) [inline pvlib/spectrum/mismatch.py:312]: When did we start leaving spaces between parameter descriptions? 😯

File: pvlib/spectrum/mismatch.py [hunk 3]
```diff
@@ -239,8 +239,9 @@ def integrate(e):
     return smm
 
 
-def spectral_factor_firstsolar(pw, airmass_absolute, module_type=None,
-                               coefficients=None, min_pw=0.1, max_pw=8):
+def spectral_factor_firstsolar(precipitable_water, airmass_absolute,
+                               module_type=None, coefficients=None,
+                               min_pw=0.1, max_pw=8):
```
Comment (AdamRJensen) [inline pvlib/spectrum/mismatch.py:244]: Perhaps a bit strange that we still use `pw` in `min_pw`  and `max_pw` - so in essence users are still required to know the acronym

**Review by AdamRJensen — COMMENTED** [review]
`precipitable_water` should be also added to the Variables and Symbols.

**Review by kandersolar — COMMENTED** [review]
> precipitable_water should be also added to the Variables and Symbols.

It was added in #1623: https://github.com/pvlib/pvlib-python/commit/50e199815b3833e6646eaebcd5afe02a0865a9ae#diff-c687fd096ceeeeaebdf4846d37b19d06e6a6c9c409a47e89bbfa3e23446ffa42

**Pull request merged**